src/titiler/xarray_api/factory.py

```diff
@@ -0,0 +1,196 @@
+                group=io_params.group,
+                decode_times=io_params.decode_times,
+            )
+
+    def statistics(self) -> None:
+        """Register /statistics and /histogram endpoints"""
+        super().statistics()
+
+        @self.router.get(
+            "/histogram",
+            response_class=JSONResponse,
+            responses={200: {"description": "Return histogram for this data variable"}},
+            response_model_exclude_none=True,
+        )
+        def histogram(
+            src_path=Depends(self.path_dependency),
+            reader_params=Depends(self.reader_dependency),
+        ):
+            with self.reader(
+                src_path=src_path,
+                variable=reader_params.variable,
+                group=reader_params.group,
+                decode_times=reader_params.decode_times,
+                datetime=reader_params.datetime,
+            ) as src_dst:
+                boolean_mask = ~np.isnan(src_dst.input)
+                data_values = src_dst.input.values[boolean_mask]
+                counts, values = np.histogram(data_values, bins=10)
+                counts, values = counts.tolist(), values.tolist()
+                buckets = list(
+                    zip(values, [values[i + 1] for i in range(len(values) - 1)])
+                )
+                hist_dict = []
+                for idx, bucket in enumerate(buckets):
+                    hist_dict.append({"bucket": bucket, "value": counts[idx]})
+                return hist_dict
+
+    def map_viewer(self) -> None:
+        """Register /map endpoints"""
+
+        @self.router.get("/{tileMatrixSetId}/map", response_class=HTMLResponse)
+        def map_viewer(
+            request: Request,
+            tileMatrixSetId: Annotated[  # type: ignore
+                Literal[tuple(self.supported_tms.list())],
+                "Identifier selecting one of the supported TileMatrixSetIds",
+            ],
+            url: Annotated[Optional[str], Query(description="Dataset URL")] = None,
+            variable: Annotated[
+                Optional[str],
+                Query(description="Xarray Variable"),
+            ] = None,
+            group: Annotated[
+                Optional[int],
+                Query(
+                    description="Select a specific zarr group from a zarr hierarchy, can be for pyramids or datasets. Can be used to open a dataset in HDF5 files."
+                ),
+            ] = None,
+            decode_times: Annotated[
+                bool,
+                Query(
+                    title="decode_times",
+                    description="Whether to decode times",
+                ),
+            ] = True,
+            drop_dim: Annotated[
+                Optional[str],
+                Query(description="Dimension to drop"),
+            ] = None,
+            datetime: Annotated[
+                Optional[str], Query(description="Slice of time to read (if available)")
+            ] = None,
```

Comment:
this is within the `reader_dependency`

Reply:
yes but I need the `url` parameter to be `Optional` for this endpoint. is there a way to override the type of a particular parameter in a dependency?

when no `url` is provided the user gets the interactive form instead of the actual map

Reply:
>  is there a way to override the type of a particular parameter in a dependency?

Yes but not if it's only for a specific endpoint, you'll need to hardcode things directly at the endpoint level

I wasn't talking about the `url` parameter but maybe a better way of doing this would be to have separate endpoints for both webpage!

Reply:
Yeah it might make more sense to have two endpoints like `/map` and `/map-builder` instead of this conditional logic.